coda-prototype/model.ts

```diff
@@ -1810,6 +1810,135 @@ class StorageManager {
     }
 }
 
+class FileIO {
+    /**
+     * Saves the given string to a file. The file is determined by a file selector UI.
+     * @param {Blob} fileContents Data to write to the file.
+     * @param {(downloadId: number) => void} onDownloadStartedHandler Function to run once the download has started
+     *                                                                successfully.
+     */
+    static saveFile(fileContents : Blob, onDownloadStartedHandler ?: (downloadId : number) => void) {
+        let url = window.URL.createObjectURL(fileContents);
+        console.log("Saving file from URL", url);
+
+        chrome.downloads.download({
+            url: url,
+            saveAs: true
+        }, onDownloadStartedHandler)
+    }
+
+    /**
+     * Exports the given dataset to file on disk.
+     * @param {Dataset} dataset Dataset to save to disk.
+     */
+    static saveDataset(dataset : Dataset) {
+        let eventJSON = {"data": [], "fields":
+            ["id", "timestamp", "owner", "data", "schemeId", "schemeName", "deco_codeValue", "deco_codeId",
+                "deco_confidence", "deco_manual", "deco_timestamp", "deco_author"]
+        }; // TODO: why are rows being referred to as 'events'?
```

Comment:
This is a mismatch between the general data structure and the way it's used in Coda. I'll explain tomorrow. Please remind me.